lib/server/services/shift/ShiftService.js

```diff
@@ -53,40 +54,48 @@ class ShiftService {
      * Return all the data for the pending shift of a specific type for a given user
      *
      * @param {UserIdentifier} shifterUserIdentifier the identifier of the user for which shift information must be retrieved
-     * @param {string} shiftType the type of the shift for which information must be retrieved
+     * @param {String} shiftType the type of the shift for which information must be retrieved
+     * @param {Date} shiftStart the start time of the shift (to check for previous information)
      * @return {Promise<object>} the shift data
      */
-    async getShiftData(shifterUserIdentifier, shiftType) {
+    async getShiftData(shifterUserIdentifier, shiftType, shiftStart = null) {
         const shift = await this.getUserPendingShiftOrFail(shifterUserIdentifier);
 
         const ret = {
             shift,
             issuesLogs: await this.getShiftIssues(shift.start, shiftType),
             typeSpecific: null,
+            previousInfo: { value: '' },
         };
 
-        // Find all EoS reports from the last 16 hours and sort by most recent first
-        const pastShift = { start: new Date() - SHIFT_DURATION * 2, end: new Date() };
-        const prevShiftIssues = (await getShiftIssues(pastShift, { include: null, exclude: [] }))
-            .reverse()
-            .sort((a, b) => {
-                new Date(b.createdAt) - new Date(a.createdAt);
-            });
-
-        // Filter for the correct tags and EoS title and select the most recent report
-        const reportFormat = `End of shift report - ${shiftType} - `;
-        const tags = getEosReportTagsByType(shiftType);
-        const allEosReports = prevShiftIssues.filter((report) =>
-            tags.every((tag) => report.tags.map((x) => x.dataValues.text).includes(tag))
-            && report.title.startsWith(reportFormat));
-        const eosReport = allEosReports.length > 0 ? allEosReports[0] : { text: '' };
-
-        // Extract the 'For next shifter' information
-        const infoFromPreviousShifterRegex = /(?<=\n\n### For next shifter\n)([\S\s]*)(?=\n\n###)/g;
-        const infoFromPreviousShifterAllMatches = eosReport.text.match(infoFromPreviousShifterRegex);
-        infoFromPreviousShifterAllMatches && infoFromPreviousShifterAllMatches[0] !== '-'
-            ? [ret.previousInfo] = infoFromPreviousShifterAllMatches
-            : ret.previousInfo = '';
+        // Find EoS reports from the last shift with the correct title
+        const timestamp = shiftStart || shift.start;
+        const pastShift = getShiftFromTimestamp(timestamp - SHIFT_DURATION);
+        const title = formatEosReportTitle(pastShift, shiftType);
+
+        const prevShiftIssues = await getShiftIssuesByTitle(pastShift, title);
+
+        if (prevShiftIssues.length !== 1) {
```

Comment:
You have two cases here: either you have no EoS report, either you have more than one